app/controllers/IncomeController.scala

```diff
@@ -257,20 +260,25 @@ class IncomeController @Inject()(personService: PersonService,
           ServiceCheckLite.personDetailsCheck {
             sendActingAttorneyAuditEvent("confirmIncomeUpdatesForEdit")
             for {
-              cachedData <- journeyCacheService.mandatoryValues(UpdateIncome_IdKey, UpdateIncome_NewAmountKey)
-              id = cachedData.head.toInt
+              cachedData <- journeyCacheService.mandatoryValues(UpdateIncome_IdKey, UpdateIncome_NewAmountKey, UpdateIncome_NameKey)
+              id :: newAmount :: pensionName :: Nil = cachedData.toList
               taxCodeIncomeDetails <- taxAccountService.taxCodeIncomes(Nino(user.getNino), TaxYear())
-              employmentDetails <- employmentService.employment(Nino(user.getNino), id)
+              employmentDetails <- employmentService.employment(Nino(user.getNino), id.toInt)
             } yield {
 
               (taxCodeIncomeDetails, employmentDetails) match {
                 case (TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]), Some(employment)) =>
-                  taxCodeIncomes.find(_.employmentId.contains(cachedData.head.toInt)) match {
+                  taxCodeIncomes.find(_.employmentId.contains(id.toInt)) match {
                     case Some(taxCodeIncome) =>
```

Comment:
Logic is fine, though I wonder if could be split up here into separate functions.  A redirect to confirm, a redirect to same and a function to decide.  Perhaps a second opinion on this?